src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtPostcodeController.java

```diff
@@ -0,0 +1,107 @@
+        @ApiResponse(code = 200, message = "Successful", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
+    public ResponseEntity<List<String>> getCourtPostcodes(@PathVariable String slug) {
+        return ok(adminService.getCourtPostcodesBySlug(slug));
+    }
+
+    /**
+     * Adds new postcodes to be handled by the court for civil service type (
+     * @param slug Court slug
+     * @param postcodes a list of postcodes to be added
+     * @return A list of postcodes created if successful, else return a list of invalid postcodes and a '400' response
+     */
+    @PostMapping(path = "/{slug}/postcodes")
+    @ApiOperation("Add new court postcodes")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Postcodes created", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 400, message = "Invalid postcodes", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role(FACT_SUPER_ADMIN)
+    public ResponseEntity<List<String>> addCourtPostcodes(@PathVariable String slug, @RequestBody List<String> postcodes) {
+        final List<String> invalidPostcodes = validationService.validatePostcodes(postcodes);
+        if (CollectionUtils.isEmpty(invalidPostcodes)) {
+            final List<String> postcodeCreated = adminService.addCourtPostcodes(slug, postcodes);
+            return created(URI.create(StringUtils.EMPTY)).body(postcodeCreated);
+        }
+        throw new InvalidPostcodeException(invalidPostcodes);
+    }
+
+    /**
+     * Deletes existing postcodes currently handled by the court for civil service type
+     * @param slug Court slug
+     * @param postcodes a list of postcodes to be deleted
+     * @return Nothing if successful, else return a list of invalid postcodes and a '400' response
+     */
+    @DeleteMapping(path = "/{slug}/postcodes")
+    @ApiOperation("Delete existing court postcodes")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Successful", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 400, message = "Invalid postcodes", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role(FACT_SUPER_ADMIN)
+    public ResponseEntity deleteCourtPostcodes(@PathVariable String slug, @RequestBody List<String> postcodes) {
+        final List<String> invalidPostcodes = validationService.validatePostcodes(postcodes);
```

Comment:
For them to be valid for deletion, do they also need to be in the database? Might be related to the point above. If we do that check before we pass it in, then thats fair enough. 

Might also be worth considering that if other projects are going to start using this service, then it might be better to put it in the API level. Could do under a different card if it adds complexity

Reply:
There is a check for postcodes not found in API AdminCourtControllerService getCourtPostcodeEntity(). No need to check that on the front end because those postcodes to be deleted are selected from the postcode checkboxes so they are guaranteed to exist from the front-end point of view